### PR TITLE
Upgrade to Atk4 3.1

### PR DESCRIPTION
**Pull request by abbadon1334**

### CODE REVIEW

File: src/Mailer/SMTP.php [hunk 1]
```diff
@@ -4,11 +4,39 @@
 
 namespace Atk4\Outbox\Mailer;
 
+use PHPMailer\PHPMailer\SMTP as PHPMailerSMTP;
+
 class SMTP extends AbstractMailer
```
Comment (mvorisek) [inline src/Mailer/SMTP.php]: should be named `Smtp` to be consistent with other naming

Comment (abbadon1334) [reply src/Mailer/SMTP.php]: done!

File: codecov.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  - tools
+comment: false
+coverage:
+  status:
+    project:
+      default:
+        target: auto
+        threshold: 0.025
+    patch: false
+    changes: false
```
Comment (mvorisek) [inline codecov.yml]: fix EOL

File: .gitignore [hunk 3]
```diff
@@ -5,3 +5,5 @@ docs/build
 .DS_Store
 .idea
 /demos/db.sqlite
```
Comment (mvorisek) [inline .gitignore:6]: is probably no longer needed

File: .gitignore [hunk 4]
```diff
@@ -5,3 +5,5 @@ docs/build
...
+/demos/db.php
+/demos/_demo-data/db.sqlite
```
Comment (mvorisek) [inline .gitignore:8]: journal should be here too

File: composer.json [hunk 5]
```diff
@@ -14,35 +14,46 @@
...
-    "php": ">=7.4.0",
-    "atk4/ui": "^2.4.0",
+    "php": ">=7.4 <8.2",
+    "atk4/data": "dev-develop",
     "html2text/html2text": "^4.3.1",
     "phpmailer/phpmailer": "^v6.4.1"
   },
+  "require-release": {
+    "php": ">=7.4 <8.2",
+    "atk4/data": "^3.1.0",
```
Comment (mvorisek) [inline composer.json]: `~` should be here

File: demos/_demo-data/create-db.php [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+echo 'GITHUB_JOB : ' . getenv('GITHUB_JOB') . PHP_EOL . PHP_EOL;
+
+if (getenv('GITHUB_JOB') === 'unit-test') {
+    echo 'skip db creation in create-db' . PHP_EOL . PHP_EOL;
+
+    return;
+}
+
+(new Migrator(new Mail($db)))->dropIfExists()->create();
```
Comment (mvorisek) [inline demos/_demo-data/create-db.php]: there should be no `dropIfExists` in CI

File: demos/_demo-data/create-db.php [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+$sqliteFile = __DIR__ . '/db.sqlite';
+if (!file_exists($sqliteFile)) {
+    new Persistence\Sql('sqlite:' . $sqliteFile);
+}
+unset($sqliteFile);
+
+/** @var Persistence\Sql $db */
+require_once __DIR__ . '/../init-db.php';
+
+echo 'GITHUB_JOB : ' . getenv('GITHUB_JOB') . PHP_EOL . PHP_EOL;
```
Comment (mvorisek) [inline demos/_demo-data/create-db.php]: `"\n"` is better than `PHP_EOL`

**Review by mvorisek — APPROVED** [review]

**Comment by abbadon1334** [comment]
All request changes done

**Pull request merged**